File: core/src/main/java/com/bibick/core/model/Item.java
package com.bibick.core.model;

import javax.persistence.*;
import java.math.BigDecimal;

/**
 * User: istrakhouski
 * Date: 12/23/13 3:17 PM
 */
@Entity
@Table(name = "item")
public class Item {

    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE)
    @Column(name = "item_id")
    private long itemId;

    @Column(name = "name")
    private String name;

    @Column(name = "description")
    private String description;

    @Column(name = "price")
    private BigDecimal price;

    public long getItemId() {
        return itemId;
    }

    public void setItemId(long item_id) {
        this.itemId = item_id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public BigDecimal getPrice() {
        return price;
    }

    public void setPrice(BigDecimal price) {
        this.price = price;
    }
}

File: web-client/src/main/webapp/resources/js/app/model/Item.js
/**
 * Created by istrakhouski on 2/14/14.
 */
Ext.define('Farm.model.Item', {
    extend: 'Ext.data.Model',
    idProperty: 'itemId',
    fields: [
        {name: 'itemId', type: 'string'},
        'name',
        'description',
        {name: 'price', type: 'string'}
    ]
});
